events/exchange.go

```diff
@@ -28,19 +28,18 @@ func NewExchange() *Exchange {
 }
 
 // Forward accepts an envelope to be direcly distributed on the exchange.
+//
+// This is useful when an event is forwaded on behalf of another namespace or
+// when the event is propagated on behalf of another publisher.
 func (e *Exchange) Forward(ctx context.Context, envelope *events.Envelope) error {
-	log.G(ctx).WithFields(logrus.Fields{
+	defer log.G(ctx).WithFields(logrus.Fields{
```

Comment:
this will be confusing if the message is printed when the actual forwarding failed.